### DISCUSSION TITLE
Array with images?

### DISCUSSION THREAD

**Discussion by jackie-chan** [apostrophecms/apostrophe#4107] [Q&A] [answered]
Sorry for potentially a silly question. I'm trying to have an image within each array entry and have that displayed on the site.
I've set up the field schema for my widget that contains an array and within the array, I'm trying to add a field to hold an image to each entry.
I had originally set it up the field as an area with an image widget, but I don't know how to show that image in the widget.html for the array element.
I then tried to set it as an attachment but again, not quite sure how to have that show up in the widget.html

Thanks!

**Comment by BoDonkey** [comment]
I'm not sure I fully understand what you are asking. I think you want to create a widget that can have multiple fields through an array, and one of those fields is an image. Is this generally correct?
I would have to see some code to understand how you were using the area option.
For the attachment, be aware that this will bring the image in without the normal editing. This may be okay for your use-case.
Another way that could be used is through a [relationship](https://v3.docs.apostrophecms.org/guide/media.html#the-relationship-field-option). Once you land on the method you want to use I can help with the HTML output.
Sharing some minimal code would help me to make sure I understand.
Cheers,
Bob

**Reply by jackie-chan** [reply]
Hey Bob,
Here's essentially what I was trying out:
`user: {
        label: 'Testimonials',
        type: 'array',
        titleField: 'userame',
        fields: {
          add: {
            userPic: {
              type: 'area',
              label: 'User Picture',
              options: {
                widgets: {
                  '@apostrophecms/image': {}
                }
              }
            },
            userame: {
              type: 'string',
              label: 'Client Name'
            }
          }
        }
      },`

But then in the widget.html, I didn't know how to loop through and show the image itself for each entry in the array.
@boutell's answer below around how to access other properties of "item", including how to reference it in the area, is what I was looking for!

**Comment by boutell** [comment, marked as answer]
Something like this should work:

{% for item in widget.arrayFieldNameHere %}
  {# Can also access other properties of "item" here if you want #}
  {% area item, "imageWidgetAreaNameHere" %}
{% endfor %}

And should permit in-context editing.

On Fri, Mar 24, 2023 at 12:17 PM Robert Means ***@***.***>
wrote:

> I'm not sure I fully understand what you are asking. I think you want to
> create a widget that can have multiple fields through an array, and one of
> those fields is an image. Is this generally correct?
> I would have to see some code to understand how you were using the area
> option.
> For the attachment, be aware that this will bring the image in without the
> normal editing. This may be okay for your use-case.
> Another way that could be used is through a relationship
> <https://v3.docs.apostrophecms.org/guide/media.html#the-relationship-field-option>.
> Once you land on the method you want to use I can help with the HTML output.
> Sharing some minimal code would help me to make sure I understand.
> Cheers,
> Bob
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/4107#discussioncomment-5420426>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27OY22G4GEXE44ZEJITW5XCJBANCNFSM6AAAAAAWGWSK5A>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: <apostrophecms/apostrophe/repo-discussions/4107/comments/5420426@
> github.com>
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Reply by jackie-chan** [reply]
Thanks so much! The missing piece was how to access other properties of "item", including how to reference it in the area, is what I was looking for!
